pytorch_lightning/utilities/jsonargparse_utils.py

```diff
@@ -0,0 +1,150 @@
+# Copyright The PyTorch Lightning team.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import os
+from typing import Type
+from jsonargparse import ArgumentParser, ActionConfigFile
+from pytorch_lightning import Trainer, LightningModule, LightningDataModule
+from pytorch_lightning.callbacks import Callback
+
+
+class LightningArgumentParser(ArgumentParser):
+    def __init__(self, *args, **kwargs):
+        """Initialize argument parser that supports configuration file input"""
+        super().__init__(*args, **kwargs)
+        self.add_argument('--cfg',
+            action=ActionConfigFile,
+            help='Path to a configuration file in json or yaml format.')
+
+
+    def add_trainer_args(
```

Comment:
do we need to add each separately, or just pass all three at once?

Reply:
I personally like them separate. But we could include one more method which calls all three.

Reply:
The idea of having separate is to give users more control for more advanced use cases, see https://github.com/mauvilsa/pytorch-lightning/blob/config_files_argparse/docs/source/trainer_cli.rst#lightningargumentparser